### PR TITLE
dotnet-sdk: Add an uninstall note and fix checkver

### LINKED ISSUE #4236: [Bug] Unable to remove dotnet-sdk
#### Current Behavior`
`scoop cleanup *` reports inability to remove a file. Using sudo doesn't help

#### Expected Behavior
Expected cleanup to remove out of date folders and files

#### Additional context/output
```
C:>scoop cleanup *
Removing dotnet-sdk: 7.0.100Remove-Item: C:\Users\bugma\scoop\apps\scoop\current\libexec\scoop-cleanup.ps1:50
Line |
  50 |          Remove-Item $dir -ErrorAction Stop -Recurse -Force
     |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Access to the path 'C:\Users\bugma\scoop\apps\dotnet-sdk\7.0.100\host\fxr\7.0.0\hostfxr.dll'
     | is denied.

C:>sudo scoop cleanup *
Removing dotnet-sdk: 7.0.100Remove-Item : Cannot remove item
C:\Users\bugma\scoop\apps\dotnet-sdk\7.0.100\host\fxr\7.0.0\hostfxr.dll: Access to the path
'hostfxr.dll' is denied.
At C:\Users\bugma\scoop\apps\scoop\current\libexec\scoop-cleanup.ps1:50 char:9
+         Remove-Item $dir -ErrorAction Stop -Recurse -Force
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : PermissionDenied: (hostfxr.dll:FileInfo) [Remove-Item], Unauthorized
   AccessException
    + FullyQualifiedErrorId : RemoveFileSystemItemUnAuthorizedAccess,Microsoft.PowerShell.Commands
   .RemoveItemCommand
```
### System details

**Windows version:** [e.g. 7, 8, 10, 11]
11
**OS architecture:** [e.g. 32bit, 64bit, arm64]
64bit
**PowerShell version:** [output of `"$($PSVersionTable.PSVersion)"`]
7.3.1
**Additional software:** [(optional) e.g. ConEmu, Git]
git version 2.39.0.windows.1
#### Scoop Configuration

```json
{
  "last_update": "2022-12-16T09:55:55.6150875+08:00",
  "scoop_repo": "https://github.com/lukesampson/scoop",
  "scoop_branch": "master"
}

```

### PR DESCRIPTION
**Pull request by StarsbySea**
Closes #4236

- [X] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/dotnet-sdk.json [hunk 1]
```diff
@@ -25,6 +25,9 @@
         "DOTNET_ROOT": "$dir",
         "MSBuildSDKsPath": "$dir\\sdk\\$version\\Sdks"
     },
+    "pre_uninstall":[
+        "Write-Host -F yellow \"If the uninstall fails with a message saying that access is denied, you may need to log out of your current account, log back in and try again. \" "
+    ],
```
Comment (rashil2000) [inline bucket/dotnet-sdk.json:30]:
```suggestion
    "pre_uninstall": "info 'If the uninstall fails with a message saying that access is denied, you may need to log out of your current account, log back in and try again.'",
```

**Comment by rashil2000** [comment]
Could you also look into the checkver issue?

File: bucket/dotnet-sdk.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "version": "7.0.200",
+    "version": "8.0.100-preview.1.23115.2",
```
Comment (rashil2000) [inline bucket/dotnet-sdk.json:2]: Preview versions are for the _Versions_ bucket. We need to look only for the stable versions here.

Comment (StarsbySea) [reply bucket/dotnet-sdk.json]: OK, I got it

**Pull request merged**